text/0134-remove-span-status.md

```diff
@@ -0,0 +1,89 @@
+# Remove Span.Status
+
+**Author:** Nikita Salnikov-Tarnovski, Splunk
+
+Remove `Span.status` and related APIs and conventions.
+
+## Motivation
+
+During numerous heated discussions in various pull requests (see
+<https://github.com/open-telemetry/opentelemetry-specification/issues/599>
+<https://github.com/open-telemetry/opentelemetry-specification/pull/697>
+<https://github.com/open-telemetry/opentelemetry-specification/pull/427>
+<https://github.com/open-telemetry/oteps/pull/69>
+<https://github.com/open-telemetry/oteps/pull/86>
+<https://github.com/open-telemetry/oteps/pull/123>)
+and during Errors Working Group meetings a lot of concerns were raised about suitability of the current `Span.Status`
+field encoded as Google RPC code for recording a result of arbitrary monitored operations from various domains
+(http requests, database queries, file system operations etc). It is hard to define clear and umambigous mapping
+from those domain to a fixed enum originated from RPC.
+
+Therefore the proposal is to remove the `Span.Status` field from Trace API and OTLP.
+
+## Internal details
+
+Several areas are affected by this change and will require a coordinated effort to make this happen.
+
+### Trace API Specification
+
+* Remove all mentions of `Status` and `Set Status` API from Span section.
+* Remove whole `Status` section.
+
+### Trace SDK Specification
+
+No changes needed.
+
+### Trace SDK exporters
+
+Remove translation rules pertaining to `Span.Status`.
+
+### Span semantic conventions
+
+* From HTTP conventions, remove `Status` section.
+* In gRPC conventions, replace the requirement to set span status with the requirement to set span attribute `grpc.status`
+to the canonical status code of the gRPC response.
+
+### OTLP data definition
+
+Rename `Span.status` field to `Span.deprecated_status` field.
+
+### Language API and SDK
+
+* Every language must remove status related code from their implementations of Trace API and SDK.
+* Every language must remove all code in their span exporters which serializes span status.
+* All instrumentation libraries, both manual and automated, must remove all calls to the removed `Span.setStatus` API.
+* All instrumentation libraries, both manual and automated, must adapt new semantic conventions for gRPC spans.
+
+### Collector
+
+Collector will have a transition period during which it has a special handling both for old data,
+which has `Span.Status` present, and new data without Status.
+This will allow for older pipelines, still relying on `Span.Status` to continue to work with as little disruption as possible.
+The exact duration of this transition period will be left for Collector's maintainers to decide.
+
+* If incoming OTLP data has `Span.Status` present, that will NOT be removed and will be passed on as-is.
+* If incoming OTLP data has `Span.Status` present and it is not equal to `OK`,
+Collector will translate that to temporary semantic attributes as follows:
```

Comment:
Is the previous point not enough? Do we need to duplicate the status as attributes?

Reply:
Yes, because new OTLP receivers will not read `status` field from the wire. But we still may want to pass this information if it was available in the Collector.

Reply:
Do we expect code behind new OTLP receivers that don't read (deprecated_)status anymore to still need status?

Reply:
Don't know, that is way we leave it there for the time being. Collector maintainers may decide when they want to remove this translation.